Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/CancelRide.hs

```diff
@@ -110,5 +132,29 @@ cancelRideHandler ServiceHandle {..} requestorId rideId req = withLogTag ("rideI
             source = source,
             reasonCode = Just reasonCode,
             driverId = mbDriverId,
+            driverCancellationLocation = currentDriverLocation,
+            driverDistToPickup = disToPickup,
             ..
           }
+
+driverDistanceToPickup ::
+  ( EncFlow m r,
+    CacheFlow m r,
+    EsqDBFlow m r,
+    Metrics.CoreMetrics m,
+    Maps.HasCoordinates tripStartPos,
+    Maps.HasCoordinates tripEndPos
+  ) =>
+  Id Merchant ->
+  tripStartPos ->
+  tripEndPos ->
+  m Meters
+driverDistanceToPickup merchantId tripStartPos tripEndPos = do
```

Comment:
change `tripStartPos` to `currentDriverLocation` and `tripEndPos` to `pickupLocation`